src/canvas/Canvas.ts

```diff
@@ -786,23 +780,25 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
    * @param {Event} e Event object fired on mouseup
    */
   __onMouseUp(e: TPointerEvent) {
-    const transform = this._currentTransform;
     this._cacheTransformEventData(e);
-    const target = this._target;
-    const isClick = this._isClick;
     this._handleEvent(e, 'up:before');
+
+    const transform = this._currentTransform;
+    const isClick = this._isClick;
+    const target = this._target;
+
     // if right/middle click just fire events and return
     // target undefined will make the _handleEvent search the target
     if (checkClick(e, RIGHT_CLICK)) {
       if (this.fireRightClick) {
-        this._handleEvent(e, 'up', RIGHT_CLICK, isClick);
+        this._handleEvent(e, 'up', RIGHT_CLICK);
       }
       return;
     }
 
     if (checkClick(e, MIDDLE_CLICK)) {
       if (this.fireMiddleClick) {
-        this._handleEvent(e, 'up', MIDDLE_CLICK, isClick);
+        this._handleEvent(e, 'up', MIDDLE_CLICK);
       }
       this._resetTransformEventData();
```

Comment:
why is this called here and not in the RIGHT_CLICK condition?

Reply:
probably a bug? each event end has to delete the old data or getPointer will be broken until the next mouse move at minimum.

Reply:
https://github.com/fabricjs/fabric.js/pull/9449/files#r1374191785